pytensor/tensor/rewriting/subtensor_lift.py

```diff
@@ -816,3 +819,79 @@ def local_subtensor_shape_constant(fgraph, node):
             return [as_tensor([1] * len(shape_parts), dtype=np.int64, ndim=1)]
     elif shape_parts:
         return [as_tensor(1, dtype=np.int64)]
+
+
+@node_rewriter([Subtensor])
+def local_subtensor_of_adv_subtensor(fgraph, node):
+    """Lift a simple Subtensor through an AdvancedSubtensor, when basic index dimensions are to the left of any advanced ones.
```

Comment:
This rewrite made me question something of the other commits. If this rewrite is lifting the subtensor out of the advanced subtensor operation, aren't some of your other rewrites actually lowering the subtensor so that it gets applied before the other ops?

Reply:
when we say lifting we mean applying closer to the inputs of the function (before other operations). What you're saying would be sinking/lowering. In general we want to apply the subtensor before other elementwise operations, so as to compute fewer things). AdvancedIndexing could be a case to lower, if we end up having more rows than before, but we don't have any rewrites to reason about that